Core/Core/AppEnvironment/SessionDefaults.swift

```diff
@@ -0,0 +1,59 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+public struct SessionDefaults {
+    private var userDefaults: UserDefaults {
+        return UserDefaults(suiteName: Bundle.main.appGroupID()) ?? .standard
+    }
+
+    private var sessionDefaults: [String: Any]? {
+        get { return userDefaults.dictionary(forKey: session.uniqueID) }
+        set { userDefaults.set(newValue, forKey: session.uniqueID) }
+    }
+
+    public let session: KeychainEntry
```

Comment:
would it be better to use `AppEnvironment.currentSession` here instead of adding another session variable?

Reply:
oh wait, i think im seeing how this works, what about passing in the session.uniqueID?

Reply:
Good call. I don't need the entire entry- I just need the uniqueID. I will switch to only taking that instead.